### ISSUE TITLE
Optional generic parameter can't use generic type in local function

### ISSUE THREAD

**Issue by twsouthwick** [dotnet/roslyn#17719]
**Version Used**: 
VS 2017 RTM

**Steps to Reproduce**:
Add the following as a local function:

```csharp
T GetField<T>(string name, T @default = default(T))
{
  return @default;
}
```

This works as a normal method, but not as a local function.

**Expected Behavior**:

Compiles and can use the local function.

**Actual Behavior**:

```
error CS0246: The type or namespace name 'T' could not be found (are you missing a using directive or an assembly reference?)
error CS1750: A value of type 'T' cannot be used as a default parameter because there are no standard conversions to type 'T'
```

**Comment by alrz** [comment]
[Seems to be fixed](https://tryroslyn.azurewebsites.net/#b:master/f:r/K4Zwlgdg5gBAygTxAFwKYFsDcBYAUAB2ACMAbMAYxnJIEMQQYBhGAbzxg5kNIpgDcA9mAAmMALIAKAJTtObXJ0UwAKjADiqZADEwqEsIA8ygHwSUAJ0iwINdKgA0KmAAFhqAGY1gJZDAC8MG6e3sgSylIyCkoc8tGK5prA5hAuQV4+OFHRAL6y0RrauvoGcMiW0KYARJVSmYq5uNlAA=).

**Comment by gafter** [comment]
@AlekseyTs Do you know if this is already fixed?

**Comment by gafter** [comment]
It looks like this may be related to the fix in https://github.com/dotnet/roslyn/pull/17294 , though I don't see any tests for this scenario.

**Comment by AlekseyTs** [comment]
No, it is not related to #17294, the problem reported here was in binding rather than in SemanticModel. It is possible that it got addressed by one of the recent changes made by @agocke. I didn't make any changes to fix this problem.

**Comment by agocke** [comment]
May have been fixed by #17014

**Comment by AlekseyTs** [comment]
We should make sure there is a unit-test for this scenario.
